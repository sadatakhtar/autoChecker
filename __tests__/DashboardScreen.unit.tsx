import React from 'react';
import renderer from 'react-test-renderer';
import Dashboard from '../screens/DashboardScreen'; // Adjust the import path according to your project structure

describe('Dashboard Component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Dashboard />).toJSON();
    expect(tree).toMatchSnapshot();
  });



  // Add more tests here as needed
});