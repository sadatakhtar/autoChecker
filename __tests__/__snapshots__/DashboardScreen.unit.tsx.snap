// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Snapshot Test renders correctly 1`] = `
<View
  style={{}}
>
  <View
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#99ccff",
        "borderColor": "#1f446d",
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 70,
        "justifyContent": "flex-start",
      }
    }
  >
    <View>
      <Text>
        LOGO
      </Text>
    </View>
    <View>
      <Text
        style={
          {
            "color": "#1f1f1f",
            "fontSize": 30,
            "fontWeight": "bold",
          }
        }
      >
        Auto
        <Text
          style={
            {
              "color": "#4c93ff",
            }
          }
        >
          C
        </Text>
        hecker
      </Text>
    </View>
  </View>
  <View
    style={
      {
        "padding": 20,
      }
    }
  >
    <Text
      style={
        {
          "color": "#1f1f1f",
          "fontSize": 24,
          "fontWeight": "bold",
        }
      }
    >
      Dashboard
    </Text>
  </View>
  <View
    style={
      {
        "borderColor": "#0360a6",
        "borderRadius": 5,
        "borderWidth": 1,
        "margin": 20,
        "padding": 20,
      }
    }
  >
    <View
      style={
        {
          "marginVertical": 6,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          {
            "fontSize": 18,
          }
        }
      >
        Simply enter the vehicle registration number, and let our powerful  algorithms do the rest. Within seconds, you'll receive a  comprehensive list of potential issues or current defects associated with the car you're interested in.
      </Text>
    </View>
    <Text
      style={
        {
          "color": "#1f1f1f",
          "fontWeight": "bold",
          "padding": 10,
        }
      }
    >
      Enter Vehicle Registration
    </Text>
    <TextInput
      onChangeText={[Function]}
      placeholder="Registration..."
      style={
        {
          "borderRadius": 5,
          "borderWidth": 1,
          "height": 40,
          "margin": 12,
          "padding": 10,
        }
      }
      value=""
    />
    <View
      style={
        {
          "flexDirection": "row",
          "justifyContent": "space-between",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#99ccff",
                "borderColor": "#1f1f1f",
                "borderRadius": 5,
                "borderWidth": 2,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
              },
              {
                "height": 40,
                "width": 150,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#1f1f1f",
                  "fontSize": 20,
                  "fontWeight": "condensedBold",
                },
                {
                  "fontSize": 14,
                },
              ]
            }
          >
            Submit
          </Text>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <View
          style={
            [
              {
                "alignItems": "center",
                "backgroundColor": "#99ccff",
                "borderColor": "#1f1f1f",
                "borderRadius": 5,
                "borderWidth": 2,
                "justifyContent": "center",
                "margin": 10,
                "padding": 10,
              },
              {
                "backgroundColor": "#ff0000",
                "height": 40,
                "width": 100,
              },
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#1f1f1f",
                  "fontSize": 20,
                  "fontWeight": "condensedBold",
                },
                {
                  "color": "#ffffff",
                  "fontSize": 14,
                },
              ]
            }
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      {
        "marginTop": 50,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <View
        style={
          [
            {
              "alignItems": "center",
              "backgroundColor": "#99ccff",
              "borderColor": "#1f1f1f",
              "borderRadius": 5,
              "borderWidth": 2,
              "justifyContent": "center",
              "margin": 10,
              "padding": 10,
            },
            undefined,
          ]
        }
      >
        <Text
          style={
            [
              {
                "color": "#1f1f1f",
                "fontSize": 20,
                "fontWeight": "condensedBold",
              },
              undefined,
            ]
          }
        >
          Back
        </Text>
      </View>
    </View>
  </View>
</View>
`;
